### PR TITLE
Implement keyword annotation feature 

### PR DESCRIPTION
**Pull request by oriole-pub**
### Description
This feature implements the highlighting of terms in course articles and a preview of more content when the cursor focuses on terms

image

### Remarks
The preview content of terminology currently only supports the use of MDX files, so each terminology should correspond to an MDX file.

**How to use term preview in Course mdx file?**

image

## Review Tips
- `Vercel Comments` feature allows reviewers to give direct feedback on **Preview Deployments**.
   Learn more about Comments in this [document](https://vercel.com/docs/concepts/deployments/comments).
   The only requirement is that all reviewers must have a Vercel account.

### CODE REVIEW

File: src/content/terms/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Terms } from '~/types/term';
+
+export const terms: Terms = {
```
Comment (Flouse) [inline src/content/terms/index.ts]: Just an idea:
I think `annotation` may be a better word than `term`
```md
## Example
- critical annotations between the lines
- annotations made in a book
```
## Why
This `keyword annotation` feature is used not only to explain new glossaries and terms, but also to explain some ideas, provide references, etc.

---
Just presenting ideas, if you stick to your wording, LGTM.

File: src/content/terms/ckb-vm.mdx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CKB VM is a crypto-agnostic virtual machine, a RISC-V instruction set based VM for executing both on-chain and off-chain code.
+
+#### Reference
+- [CKB-VM from RFCS](https://docs.nervos.org/docs/basics/glossary#general-glossary)
```
Comment (Flouse) [inline src/content/terms/ckb-vm.mdx]:
```suggestion
- [CKB-VM RFC (Final)](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0003-ckb-vm/0003-ckb-vm.md)
```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: src/content/courses/basic-theory/chapter_1.mdx [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
 
-To understand `CKB`, we must leave behind all the complex concepts and capture the very essence: it's all about cells and the transformation of cells.
-
-`Cell` is the basic unit of CKB, similar to a cell in the human body. All the cells constitute the general state of the entire CKB blockchain. When we initiate a transaction on the blockchain, thus making a state change, in the end, for CKB, it is nothing more than spending some cells while creating some new ones, no matter how complicated the transaction and the state change are. This process is the same as the [Bitcoin UTXO](https://developer.bitcoin.org/glossary.html?highlight=utxo).
+:term[Cell] is the basic unit of :term[CKB], similar to a cell in the human body. All the cells constitute the general state of the entire CKB blockchain. When we initiate a transaction on the blockchain, thus making a state change, in the end, for CKB, it is nothing more than spending some cells while creating some new ones, no matter how complicated the transaction and the state change are. This process is the same as the [Bitcoin UTXO](https://developer.bitcoin.org/glossary.html?highlight=utxo).
```
Comment (Flouse) [inline src/content/courses/basic-theory/chapter_1.mdx]:
```suggestion
:term[Cell] is the basic unit of CKB, similar to a cell in the human body. All the cells constitute the general state of the entire CKB blockchain. When we initiate a transaction on the blockchain, thus making a state change, in the end, for CKB, it is nothing more than spending some cells while creating some new ones, no matter how complicated the transaction and the state change are. This process is the same as the [Bitcoin UTXO](https://developer.bitcoin.org/glossary.html?highlight=utxo).
```

File: src/content/terms/ckb.mdx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+All data on Nervos CKB is stored in cells. Cells are the primary state units in CKB, within them users can include arbitrary states.
```
Comment (Flouse) [inline src/content/terms/ckb.mdx]:
```suggestion
An abbreviation which can have different meanings depending on the context:
```

**Review by Flouse — CHANGES_REQUESTED** [review]

**Review by RetricSu — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

**Comment by Flouse** [comment]
> image

The link in the `annotation view` should open a new browser tab by default.

**Pull request merged**